subprojects/core/src/main/java/org/gradle/internal/progress/TaskOutcomeStatisticsFormatter.java

```diff
@@ -20,22 +20,29 @@
 import org.gradle.api.tasks.TaskState;
 
 public class TaskOutcomeStatisticsFormatter {
-    private int executedTasksCount;
+    private int avoidedTasksCount;
     private int allTasksCount;
 
-    public String incrementAndGetProgress(TaskState state) {
+    public String incrementAndGetProgress(final TaskState state) {
```

Comment:
I think I just realized that this is probably not thread-safe. avoidedTasksCount and allTasksCount are updated independently, but `BuildProgressLogger.afterExecute` is probably called concurrently sometimes?